AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/Storage/CascadeDeleteOperation.swift

```diff
@@ -486,6 +508,12 @@ extension CascadeDeleteOperation {
             }
         }
     }
+    
+    enum isDeveloperDefined: ModelName {
+        case ModelSyncMetadata = "ModelSyncMetadata"
+        case MutationEvent = "MutationEvent"
+        case MutationSyncMetadata = "MutationSyncMetadata"
+    }
```

Comment:
should use the system Model type's themselves instead of hardcoding the system model names, what if they change? this code will continue to compile. Using the types give you access to the model's name, for example, the MutationEvent type should have a `modelName` or `name` property that should be used for model name string checking

Reply:
based on my comment about creating a computed property that returns whether it is a developer defined model or not should render this enum unused. We can probably use a Set or an Array to hold the system model names and use that to check when checking if `modelSchema.name` is a developer defined model or not.